clap_complete/src/shells/bash.rs

```diff
@@ -164,29 +164,48 @@ fn option_details_for_path(cmd: &Command, path: &str) -> String {
     let mut opts = vec![String::new()];
 
     for o in p.get_opts() {
+        let compopt = match o.get_value_hint() {
+            ValueHint::FilePath => Some("compopt -o filenames"),
+            _ => None,
+        };
+
         if let Some(longs) = o.get_long_and_visible_aliases() {
             opts.extend(longs.iter().map(|long| {
-                format!(
-                    "--{})
-                    COMPREPLY=({})
-                    return 0
-                    ;;",
-                    long,
-                    vals_for(o)
-                )
+                let mut v = vec![
+                    format!("--{})", long),
+                    format!("COMPREPLY=({})", vals_for(o)),
+                ];
+
+                if let Some(copt) = compopt {
+                    v.extend([
+                        r#"if [ "${BASH_VERSINFO[0]:-0}" -ge 4 ]; then"#.to_string(),
```

Comment:
This compares the major version.  Do we also need to check the minor version like #5278 does?

Reply:
I think we don't need to check the minor version because `compopt` was introduced in bash 4.**0**. (In other words, it is sufficient to check the major version only because the minor version is always greater than or equal to 0.)

By the way, should I implement this comparison in the same way of #5278 for consistency?
I wrote this in POSIX-compatible way considering #5011, but it requires much effort to fix everything...

Reply:
I can go either way.  Without us having explicit testing for posix compatibility in our generated output, we'll regularly mess it up so I don't think its worth it until someone intentionally tries to support it.

Reply:
OK. I'll go the same way as #5278.